test/fixtures/bn.json

```diff
@@ -0,0 +1,105 @@
+{
+  "valid": [
+    {
+      "dec": "0",
+      "hex": ""
+    },
+    {
+      "dec": "1",
+      "hex": "01"
+    },
+    {
+      "dec": "-1",
+      "hex": "ff"
+    },
+    {
+      "dec": "127",
+      "hex": "7f"
+    },
+    {
+      "dec": "-127",
+      "hex": "81"
+    },
+    {
+      "dec": "128",
+      "hex": "80"
+    },
+    {
+      "dec": "-128",
+      "hex": "80"
+    },
+    {
+      "dec": "255",
+      "hex": "ff"
+    },
+    {
+      "dec": "-255",
+      "hex": "ff01"
+    },
+    {
+      "dec": "16300",
+      "hex": "3fac"
+    },
+    {
+      "dec": "-16300",
+      "hex": "c054"
+    },
+    {
+      "dec": "62300",
+      "hex": "f35c"
+    },
+    {
+      "dec": "-62300",
+      "hex": "ff0ca4"
+    },
+    {
+      "dec": "158798437896437949616241483468158498679",
+      "hex": "77777777777777777777777777777777"
+    },
+    {
+      "dec": "115792089237316195423570985008687907852837564279074904382605163141518161494336",
+      "hex": "fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364140"
+    },
+    {
+      "dec": "48968302285117906840285529799176770990048954789747953886390402978935544927851",
+      "hex": "6c4313b03f2e7324d75e642f0ab81b734b724e13fec930f309e222470236d66b"
+    }
+  ],
+  "invalid": [
```

Comment:
@dcousens I don't even use `.toString()`/`parse` internally, actually. I wonder if the regexp in assert before entering the `_parseHex` and `_parseBase` would be enough? Though, it might be a bit more complicated for `_parseBase`, since the character set may different between different bases.